lib/dialects/mssql/index.js

```diff
@@ -213,8 +213,12 @@ Object.assign(Client_MSSQL.prototype, {
   // Get a raw connection, called by the `pool` whenever a new
   // connection needs to be added to the pool.
   acquireRawConnection() {
-    return new Bluebird((resolver, rejecter) => {
-      const settings = Object.assign({}, this.connectionSettings);
+    const client = this;
+    return new Bluebird(async (resolver, rejecter) => {
+      const settings = await client
+        .connectionConfigProvider()
+        .resolveConnectionConfig();
+      // const settings = Object.assign({}, this.connectionSettings);
```

Comment:
is this needed?

Reply:
Yes, it is needed. if the config is passed as an async function, then this will cause a problem without resolving the config.

Also,  this line 
```
const settings = Object.assign({}, this.connectionSettings);
```
is not needed any more as `ConnectionConfigProvider` is taking the responsibility for this.

I will clarify the need for this PR again in case I miss any clarification.

The main purpose is we need to fetch the config of the DB from an async store (AWS SSM as an example) and this requires passing the config as an async function in order to wait for the config to be loaded from the promise.

Travis 1st stage passes but the other stages failed and I believe it is because of oracledb dialect which fails only in one of the test cases. Can you have a look please and help?

Reply:
Can you please delete the unneeded commented out code from PR?

Reply:
re: Oracle

this seems directly related:
```
1) Oracledb driver tests
       OracleDb externalAuth
         externalAuth and connectString should be sent to the getConnection:
     AssertionError: expected getConnection to have been called exactly once, but it was called 0 times
      at Context.<anonymous> (test/unit/dialects/oracledb.js:36:25)
```

I believe you'd need to update some of mocks in oracle tests, as call hierarchy has obviously changed.

Reply:
re: Oracledb

I believe the issue is because of switching the bludbird callback to async in line 79

```
const asyncConnection = new Bluebird(async function(resolver, rejecter) {
//...res of the code
}
```
where the linter is showing an error at line 105
```
    client.driver.fetchAsString = client.fetchAsString;
```
error:
`possible race condition client.driver.fetchAsString might be reassinged based on an outdated value of client.driver.fetchAsString`

need to debug the code but the issue I can't get the tests working on my machine due to insufficient storge  issue (mac 128 GB)

any ideas?

Reply:
I assume this is still relevant?

Reply:
@kibertoad I've checked the tests. only Oracle dialect fails. and honestly, I have no idea why. can you please help? I will do one more round to try to find the problem

Reply:
>externalAuth and connectString should be sent to the getConnection
add await in test 
```
-     knexInstance.client
+    await knexInstance.client
      .acquireRawConnection()
      .then(function(resolve) {}, function(reject) {});
```

Reply:
@maximelkin Which test in which file?